src/cacheServerClient/cacheServerClient.ts

```diff
@@ -96,20 +100,28 @@ export class CacheServerClient implements ICacheServerClient {
     return data.orgs;
   }
 
-  async getOrganizationsBySearchPhrase({ search }: { search: string }) {
-    const { data } = await this.httpClient.get<IOrganization[]>(
-      `/namespace/search/${search}?type=Org`
-    );
-    return data;
-  }
-
-  async getApplicationsBySearchPhrase({ search }: { search: string }) {
-    const { data } = await this.httpClient.get<IApp[]>(`/namespace/search/${search}?type=App`);
-    return data;
-  }
-
-  async getRolesBySearchPhrase({ search }: { search: string }) {
-    const { data } = await this.httpClient.get<IRole[]>(`/namespace/search/${search}?type=Role`);
+  async getNamespaceBySearchPhrase({
+    types,
+    search
+  }: {
+    types?: ("App" | "Org" | "Role")[];
+    search: string;
+  }) {
+    if (types && types.length > 0) {
+      const { data } = await this.httpClient.get<IOrganization[] | IApp[] | IRole[]>(
+        `/namespace/search/${search}`,
+        {
+          params: {
+            types
+          },
+          paramsSerializer: params => {
+            return stringify(params, { arrayFormat: "brackets" });
+          }
+        }
+      );
+      return data;
+    }
+    const { data } = await this.httpClient.get<IOrganization[]>(`/namespace/search/${search}`);
```

Comment:
Maybe still one spot that needs `IOrganization[] | IApp[] | IRole[]`. Regardless, PR can be merged as is or after it's fixed.